src/recipes/heartbeat-by-user-first-impression/utils.js

```diff
@@ -31,9 +31,17 @@ var getEngagementUrl = function(obj, engagementRules, rng=Math.random()) {
   let ruleIndex = firstMatch(obj, rules);
   if (ruleIndex >= 0) {
     let possibles = engagementRules[ruleIndex];
-    let url = cutBreaks(possibles.urls, possibles.breaks, rng)
+
+    let url = cutBreaks(possibles.urls, possibles.breaks, rng);
     if (url && (url.indexOf("qsurvey") >= 0)) {
-      url = url + `?source=heartbeat&surveyversion=${obj.VERSION}&updateChannel=${obj.updateChannel}&fxVersion=${obj.fxVersion}`
+
+      let additional = ""; //TODO: Remove this hacky, temp code
```

Comment:
oooh, I really hate this :)  

I think BETTER is to add the query arg to the original url in the CONFIG file.  That should Just Work.  See comment above.

If you HAVE to do it this way, at least put a big old comment with the bug number.  "special cased for this one survey"